src/audiovideocontroller/DefaultAudioVideoController.ts

```diff
@@ -564,7 +564,7 @@ export default class DefaultAudioVideoController
     if (!audioStream || audioStream.getAudioTracks().length < 1) {
       throw new Error('could not acquire audio track');
     }
-
+    this._realtimeController.realtimeSetLocalAudioInput(audioStream);
```

Comment:
Suggest moving this to after the following `awaits` to `replaceAudioTrack` to ensure that the realtime controller is only updated if the track is actually replaced:

```
this._realtimeController.realtimeSetLocalAudioInput(audioStream);
this.meetingSessionContext.activeAudioInput = audioStream;
```

Reply:
When we reach this point in the code, the local audio input stream (or active audio input in deviceController) already switches to the new stream right. So I thought it would make sense to do it here

Reply:
What matters however is the track that is attached to the peer connection.

Reply:
@aburnson  Correct me if I am wrong but let's say the replaceTrack failed for some reason and we did not update realtime local audio input. At this point the local audio input device already switches to a new device (Let's say to device B from device A). Since the realtime local audio input is still A, realtime Mute will try to mute that instead of B.
Isn't that the bug we are fixing here?